chromedp.go

```diff
@@ -244,7 +244,7 @@ func FromContext(ctx context.Context) *Context {
 // underlying errors happening during cancellation.
 func Cancel(ctx context.Context) error {
 	c := FromContext(ctx)
-	if c == nil {
+	if c == nil || c.cancel == nil {
```

Comment:
I think we only need to add a check before `c.cancel()` on line 257.

```diff
    } else {
+       // c.cancel is nil when Cancel is wrongly called with a context returned
+       // from chromedp.NewRemoteAllocator or chromedp.NewRemoteAllocator.
+       if c.cancel == nil {
+           return ErrInvalidContext
+       }
        c.cancel()
        c.closedTarget.Wait()
    }
```

Reply:
But `c.cancel()` is called again in line 275, so the error could occur again there.

Furthermore, this error no longer occurs if cancel is passed to the context in chromedp.NewRemoteAllocator or chromedp.NewExecAllocator.

And the correct comment would be:
```
// c.cancel is nil when Cancel is wrongly called with a context returned 
// from chromedp.NewRemoteAllocator or chromedp.NewExecAllocator
```

Reply:
You are right, `c.cancel` is always called no matter whether `graceful` is true or false. Then we should move this check to line 247 as what you have done.

Good catch for the comment message. Lets put `chromedp.NewExecAllocator` first.

```
// c.cancel is nil when Cancel is wrongly called with a context returned 
// from chromedp.NewExecAllocator or chromedp.NewRemoteAllocator
```

Reply:
> Furthermore, this error no longer occurs if cancel is passed to the context in chromedp.NewRemoteAllocator or chromedp.NewExecAllocator.

It just hides the issue. If we do this, it will change the behavior of the `Cancel` method. If you think that we should change the behavior, please document the new behavior.

Reply:
I think you are right, I have reverted the changes.